pkg/trace/api/api.go

```diff
@@ -390,6 +402,38 @@ func (r *HTTPReceiver) rateLimited(n int64) bool {
 	return !r.RateLimiter.Permits(n)
 }
 
+// StatsProcessor implementations are able to process incoming client stats.
+type StatsProcessor interface {
+	// ProcessStats takes a stats payload and consumes it. It is considered to be originating
+	// from the given lang.
+	ProcessStats(p pb.ClientStatsPayload, lang string)
+}
+
+// handleStats handles incoming stats payloads.
+func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.receiver.stats_process_ms", time.Now())
+
+	ts := r.tagStats(v05, req)
+	rd := NewLimitedReader(req.Body, r.conf.MaxRequestBytes)
+	metrics.Count("datadog.trace_agent.receiver.stats_payload", 1, ts.AsTags(), 1)
+	if v := req.Header.Get("Content-Length"); v != "" {
+		d, err := strconv.ParseInt(v, 10, 64)
+		if err == nil {
+			metrics.Count("datadog.trace_agent.receiver.stats_bytes", d, ts.AsTags(), 1)
+		}
+	}
+
+	req.Header.Set("Accept", "application/msgpack")
+	var in pb.ClientStatsPayload
+	if err := msgp.Decode(rd, &in); err != nil {
+		httpDecodingError(err, []string{"handler:stats", "codec:msgpack", "v:v0.5"}, w)
+		return
+	}
+	metrics.Count("datadog.trace_agent.receiver.stats_buckets", int64(len(in.Stats)), ts.AsTags(), 1)
```

Comment:
I can't figure out what writes the http response in this function, how does it work?

Reply:
There is no need to, the server automatically responds 200 OK if the user doesn't ([example](https://play.golang.org/p/plxicRqw1-U))